### PR TITLE
App 15094 swirl components in react

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary 
(generated by ChatGPT)

### Components
Summary of Code Changes:
1. **DocumentationHeader.tsx**:
   - Imported the `Heading` component and replaced the `<h1>` element with it.
   - Imported the `SwirlText` component and replaced the `<p>` element with it.

2. **Parameter.tsx**:
   - Imported the `SwirlText` component and wrapped the `description` prop with it.

3. **CategoryNav.tsx**:
   - Refactored the rendering logic of sub-items in the category navigation.
   - Added animation and transitions for sub-item animations.
   - Updated the styling and structure of navigation items and sub-items.

4. **DocumentationLayout.tsx**:
   - Removed the `footer` prop.

5. **Tags/index.tsx**:
   - Added a new function to map HTTP methods to tag content.
   - Updated the `Tag` component to handle the new `httpTag` prop.

6. **Headers/index.tsx**:
   - Added the `Heading` and `LinkedHeading` components.

7. **Text/index.tsx**:
   - Added the `SwirlText` component.

8. **index.ts**:
   - Exported the `Headers` and `Text` components from the `swirl-recreations` module.

### Lib
Summary of Code Changes:

1. **index.ts**:
   - Removed the import statement for `componentsJSON`.
   - Used `require` to import `componentsJSON` from `@getflip/swirl-components/components.json`.

2. **docs.model.ts**:
   - Removed type annotations for the `description` property of `ApiDocumentation`.
   - Changed the type of `description` to `MDXRemoteSerializeResult`.

3. **singleDoc.ts**:
   - Updated the return type of the `generateMdxFromDocumentation` function to `Promise<MDXRemoteSerializeResult>`.

### Pages Directory
Summary of Code Changes:

1. **[apiSpec].tsx**:
   - Renamed the `getSpecData` function to `generateSpecData`.
   - Added an import statement for `useRouter` from "next/router".
   - Updated the return type of the `generateSpecData` function to `Promise<ApiDocumentation>`.
   - Removed the unused import statements for `Heading`, `LinkedHeading`, `SwirlText`, and `Tag`.
   - Removed the unused import statement for `CodePreview`.
   - Replaced the deprecated `mdxContent` prop with `document.description` in the `props` object for the `<DocumentationLayout.MDX />` component.
   - Removed the unused `footer` prop.

2. **index.tsx**:
   - Removed the unused import statement for `CodePreview`.
   - Removed the unused `footer` prop.

3. **[componentDoc].tsx**:
   - Removed the unused `footer` prop.

4. **[tokenDoc].tsx**:
   - Removed the unused `footer` prop.

### CODE REVIEW

File: apps/swirl-docs/src/components/Tags/index.tsx [hunk 1]
```diff
@@ -4,6 +4,24 @@ import { FunctionComponent } from "react";
...
+    case "GET":
+      return "GET";
+    case "POST":
+      return "POST";
+    case "PUT":
+      return "PUT";
+    case "DELETE":
+      return "DEL";
+    default:
+      return "GET";
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Tags/index.tsx:23]: **issue:** Would return an incorrect label for e.g. methods "OPTIONS", "HEAD" or "TRACE".

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 2]
```diff
@@ -156,8 +164,16 @@ export default function Document({ document }: { document: ApiDocumentation }) {
...
             components: {
-              h1: (props) => <h2 className="text-2xl font-bold" {...props} />,
-              h2: (props) => <h3 className="text-xl font-bold" {...props} />,
+              h1: (props) => (
+                <LinkedHeading href="#">
+                  <Heading level={1} {...props} />
+                </LinkedHeading>
+              ),
+              h2: (props) => (
+                <LinkedHeading href="#">
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:173]: **issue:** href should link to the correct URL fragment

File: apps/swirl-docs/src/components/swirl-recreations/Headers/index.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import React, {
```
[the remark this answers is not in the document]
Comment (Sqrrl) [reply apps/swirl-docs/src/components/swirl-recreations/Headers/index.tsx:1]: Alright. Could we just call them "Text" and "Heading" then, without the "Swirl"?

[the remark this answers is not in the document]
Comment (webspaceadam) [reply apps/swirl-docs/src/components/swirl-recreations/Headers/index.tsx:1]: sure!

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 4]
```diff
@@ -156,8 +164,16 @@ export default function Document({ document }: { document: ApiDocumentation }) {
           mdxContent: {
             document: document.description,
             components: {
-              h1: (props) => <h2 className="text-2xl font-bold" {...props} />,
-              h2: (props) => <h3 className="text-xl font-bold" {...props} />,
+              h1: (props) => (
+                <LinkedHeading href="#">
```
[the remark this answers is not in the document]
Comment (webspaceadam) [reply apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:168]: removed the linked heading. it is unnecessary anyway

**Comment by webspaceadam** [comment]
@Sqrrl 
image
The second issue is caused by the codepreview component. as it got a redesign and needs further development i would handle this in its own ticket. 

Furthermore i will take the first issue to discussions with design.

File: apps/swirl-docs/src/components/Tags/index.tsx [hunk 5]
```diff
@@ -19,6 +19,14 @@ export function mapHttpMethodToTagContent(
...
     case "DELETE":
       return "DEL";
+    case "OPTIONS":
+      return "OPT";
+    case "PATCH":
+      return "PAT";
+    case "HEAD":
+      return "HEAD";
+    case "TRACE":
+      return "TRA";
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Tags/index.tsx:29]: **issue:** CONNECT is missing.

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 6]
```diff
@@ -164,26 +159,18 @@ export default function Document({ document }: { document: ApiDocumentation }) {
...
                   <a {...props} />
-                  <SwirlIconOpenInNew className="ml-1" size={20} />
+                  <i className="swirl-icons-OpenInNew28 text-[1.25rem] ml-1"></i>
                 </span>
               ),
               ul: (props) => (
                 <ul className="mb-4 leading-line-height-xl" {...props} />
               ),
-              p: (props) => <SwirlText {...props} />,
+              // p: (props) => <Text {...props} />,
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:173]: **question:** Why commented out?

**Review by Sqrrl — CHANGES_REQUESTED** [review]
I would still argue that the "swirl-recreations" directory is unnecessary. Could live with it, though. ;)

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx (hunk 6 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:173]: not intended

**Pull request merged**